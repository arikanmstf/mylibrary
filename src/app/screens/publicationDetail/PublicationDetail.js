/**
 * Screen Template By => create-module script
 * @version 1.0.1
 *
 */

// @flow
import * as React from 'react';
import logger from 'helpers/logger';
import type { PublicationDetailProps } from './PublicationDetailTypes';

// eslint-disable-next-line react/prefer-stateless-function
class PublicationDetail extends React.Component<PublicationDetailProps> {
  componentDidMount() {
    const { fetchPublication, match: { params: { id } }, card } = this.props;

    if (!card || (card && card.id !== +id)) {
      fetchPublication(id);
    }
  }

  render() {
    const {
      Screen,
      Header,
      Page,
      CardDetail,
      card,
    } = this.props;
    logger.log('render: PublicationDetail');

    return (
      <Screen>
        <Header />
        <Page>
          <CardDetail
            card={card}
            isDetailed
          />
        </Page>
      </Screen>
    );
  }
}

export default PublicationDetail;
