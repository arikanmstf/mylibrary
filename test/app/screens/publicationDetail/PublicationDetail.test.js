/**
 * Screen Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import PublicationDetail from 'screens/publicationDetail/PublicationDetail';

describe('test/app/screens/publicationDetail/PublicationDetail.test.js', () => {
  const props = {
    match:
      {
        params: {
          id: 1,
        },
      },
    fetchPublication: jest.fn(),
  };

  beforeAll(() => {
    props.fetchPublication.mockClear();
  });

  it('Render', () => {
    const wrapper = shallow(<PublicationDetail {...props} />);
    const { fetchPublication } = props;

    expect(wrapper).toMatchSnapshot();
    expect(fetchPublication).toBeCalledWith(1);
  });
});
