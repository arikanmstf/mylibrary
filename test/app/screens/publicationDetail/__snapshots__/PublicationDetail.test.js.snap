// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/screens/publicationDetail/PublicationDetail.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PublicationDetail
    fetchPublication={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          back={true}
          title="HOME_TITLE"
        />,
        <UNDEFINED>
          <UNDEFINED
            isDetailed={true}
          />
        </UNDEFINED>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "back": true,
          "title": "HOME_TITLE",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <UNDEFINED
            isDetailed={true}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": undefined,
            "isDetailed": true,
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        "type": undefined,
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            back={true}
            title="HOME_TITLE"
          />,
          <UNDEFINED>
            <UNDEFINED
              isDetailed={true}
            />
          </UNDEFINED>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "back": true,
            "title": "HOME_TITLE",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <UNDEFINED
              isDetailed={true}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "card": undefined,
              "isDetailed": true,
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
          "type": undefined,
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
